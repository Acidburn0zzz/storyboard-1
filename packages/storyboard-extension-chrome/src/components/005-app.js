/* eslint-env browser */

/* eslint-disable react/no-string-refs */

import React from 'react';
import * as ReactRedux from 'react-redux';
import {
  Floats, Notifications,
  Hints, hintDefine, hintShow,
  LargeMessage,
  Spinner,
  isDark,
} from 'giu';
import tinycolor from 'tinycolor2';
import pick from 'lodash/pick';
import Toolbar from './015-toolbar';
import Story from './020-story';
// let ReduxDevTools;
// if process.env.NODE_ENV isnt 'production' {
//   ReduxDevTools = require('../components/990-reduxDevTools')
// }

require('./app.sass');

const mapStateToProps = (state) => ({
  fRelativeTime: state.settings.timeType === 'RELATIVE',
  cxState: state.cx.cxState,
  mainStory: state.stories.mainStory,
  colors: pick(state.settings, [
    'colorClientBg', 'colorServerBg', 'colorUiBg',
    'colorClientFg', 'colorServerFg', 'colorUiFg',
    'colorClientBgIsDark', 'colorServerBgIsDark', 'colorUiBgIsDark',
  ]),
});

class App extends React.PureComponent {
  static propTypes = {
    // From Redux.connect
    fRelativeTime: React.PropTypes.bool.isRequired,
    cxState: React.PropTypes.string.isRequired,
    mainStory: React.PropTypes.object.isRequired,
    colors: React.PropTypes.object.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      seqFullRefresh: 0,
      timeRef: null,
    };
  }

  componentDidMount() {
    this.timerFullRefresh = setInterval(this.fullRefresh, 30e3);
    window.addEventListener('scroll', this.onScroll);
    this.showHint();
  }

  componentWillUnmount() {
    clearInterval(this.timerFullRefresh);
    this.timerFullRefresh = null;
    window.removeEventListener('scroll', this.onScroll);
  }

  componentDidUpdate() {
    if (this.fAnchoredToBottom) {
      window.scrollTo(0, document.body.scrollHeight);
    }
    this.showHint();
  }

  // -----------------------------------------------------
  fullRefresh = () => {
    if (!this.props.fRelativeTime) return;
    this.setState({
      seqFullRefresh: this.state.seqFullRefresh + 1,
    });
  }

  onScroll = () => {
    const bcr = this.refs.outer.getBoundingClientRect();
    this.fAnchoredToBottom = bcr.bottom - window.innerHeight < 30;
  }

  // -----------------------------------------------------
  render() {
    let reduxDevTools;
    const { cxState, colors } = this.props;
    // if (process.env.NODE_ENV isnt 'production') {
    //   reduxDevTools = <ReduxDevTools />
    // }
    const fConnected = cxState === 'CONNECTED';
    return (
      <div ref="outer" id="appRoot" style={style.outer(colors)}>
        <Floats />
        <Notifications />
        <Hints />
        {!fConnected && this.renderConnecting()}
        {fConnected && <Toolbar colors={colors} />}
        {fConnected && this.renderStories()}
        {reduxDevTools}
      </div>
    );
  }

  renderStories() {
    return (
      <div style={style.stories}>
        <Story
          story={this.props.mainStory}
          level={0}
          seqFullRefresh={this.state.seqFullRefresh}
          timeRef={this.state.timeRef}
          setTimeRef={this.setTimeRef}
          colors={this.props.colors}
        />
      </div>
    );
  }

  renderConnecting() {
    return (
      <LargeMessage style={style.largeMessage(this.props.colors)}>
        <div><Spinner /> Connecting to Storyboard...</div>{' '}
        <div>Navigate to your Storyboard-equipped app (and log in if required)</div>
      </LargeMessage>
    );
  }

  // -----------------------------------------------------
  setTimeRef = (timeRef) => { this.setState({ timeRef }); }

  showHint() {
    if (this.props.cxState !== 'CONNECTED') return;
    if (this.fAttempted) return;
    this.fAttempted = true;
    const elements = () => {
      const out = [];
      const nodeToolbar = document.getElementById('sbToolbar');
      if (nodeToolbar) {
        const wWin = window.innerWidth;
        const bcr = nodeToolbar.getBoundingClientRect();
        let x;
        x = 50;
        const y = 80;
        if (wWin > 500) {
          out.push({
            type: 'LABEL', x, y, align: 'left',
            children: <span>Settings, collapse/expandâ€¦<br />that sort of stuff</span>,
            style: { width: 150 },
          });
          out.push({
            type: 'ARROW', from: { x: x - 5, y },
            to: { x: 20, y: bcr.bottom - 2 },
          });
        }
        // ----------
        x = wWin - 80;
        out.push({
          type: 'LABEL', x, y, align: 'right',
          children: <span>Maybe you need to log in to see <span style={{ color: 'yellow' }}>server logs</span></span>,
          style: { width: 150 },
        });
        out.push({
          type: 'ARROW', from: { x: x + 5, y },
          to: { x: wWin - 50, y: bcr.bottom - 2 },
          counterclockwise: true,
        });
      }
      // ----------
      {
        const x = 50;
        const y = 230;
        out.push({
          type: 'LABEL', x, y, align: 'left',
          children: (
            <span>
              <span style={{ color: '#4df950' }}>Click</span> on timestamps to toggle: local, UTC, relative to now
              <br />
              <span style={{ color: '#4df950' }}>Right-click</span> to set a reference timestamp
            </span>
          ),
          style: { width: 2000 },
        });
        out.push({
          type: 'ARROW', from: { x: x - 5, y },
          to: { x: 20, y: y + 40 },
          counterclockwise: true,
        });
      }
      return out;
    };
    hintDefine('main', { elements, closeLabel: 'Enjoy!' });
    hintShow('main');
  }
}

// -----------------------------------------------------
const style = {
  outer: (colors) => ({
    height: '100%',
    backgroundColor: colors.colorUiBg,
    color: colors.colorUiFg,
  }),
  largeMessage: (colors) => {
    let color = tinycolor(colors.colorUiFg);
    color = isDark(colors.colorUiFg) ? color.lighten(20) : color.darken(20);
    return { color: color.toRgbString() };
  },
  stories: { padding: 4 },
};

// -----------------------------------------------------
const connect = ReactRedux.connect(mapStateToProps);
export default connect(App);
export { App as _App };
