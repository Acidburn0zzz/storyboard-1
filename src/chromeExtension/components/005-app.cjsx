_                 = require '../../vendor/lodash'
React             = require 'react'
ReactRedux        = require 'react-redux'
{
  Floats,
  LargeMessage,
  Spinner,
}                 = require 'giu'
Toolbar           = require './015-toolbar'
Story             = require './020-story'
if process.env.NODE_ENV isnt 'production'
  ReduxDevTools   = require '../components/990-reduxDevTools'

require './app.sass'
require 'font-awesome/css/font-awesome.css'

mapStateToProps = (state) -> 
  fRelativeTime:  state.settings.timeType is 'RELATIVE'
  cxState:        state.cx.cxState
  mainStory:      state.stories.mainStory

App = React.createClass
  displayName: 'App'

  #-----------------------------------------------------
  propTypes:
    # From Redux.connect
    fRelativeTime:          React.PropTypes.bool.isRequired
    cxState:                React.PropTypes.string.isRequired
    mainStory:              React.PropTypes.object.isRequired
  getInitialState: ->
    seqFullRefresh:         0

  #-----------------------------------------------------
  componentDidMount: -> 
    @timerFullRefresh = setInterval @fullRefresh, 30e3
    window.addEventListener 'scroll', @onScroll

  componentWillUnmount: ->
    clearInterval @timerFullRefresh
    @timerFullRefresh = null
    window.removeEventListener 'scroll', @onScroll

  componentDidUpdate: ->
    if @fAnchoredToBottom
      window.scrollTo 0, document.body.scrollHeight

  fullRefresh: -> 
    return if not @props.fRelativeTime
    @setState {seqFullRefresh: @state.seqFullRefresh + 1}

  onScroll: ->
    bcr = @refs.outer.getBoundingClientRect()
    @fAnchoredToBottom = (bcr.bottom - window.innerHeight) < 30

  #-----------------------------------------------------
  render: -> 
    reduxDevTools = undefined
    ## if process.env.NODE_ENV isnt 'production'
    ##   reduxDevTools = <ReduxDevTools/>
    fConnected = @props.cxState is 'CONNECTED'
    <div ref="outer" id="appRoot" style={_style.outer}>
      <Floats />
      {if not fConnected then @renderConnecting()}
      {if fConnected then <Toolbar/>}
      {if fConnected then @renderStories()}
      {reduxDevTools}
    </div>

  renderStories: ->
    <div style={_style.stories}>
      <Story 
        story={@props.mainStory} 
        level={0} 
        seqFullRefresh={@state.seqFullRefresh}
      />
    </div>

  renderConnecting: ->
    <LargeMessage>
      <div><Spinner /> Connecting to Storyboard... </div>
      <div>Navigate to your Storyboard-equipped app (and log in if required)</div>
    </LargeMessage>

#-----------------------------------------------------
_style = 
  outer: 
    backgroundColor: 'white'
    backgroundOpacity: 0.5
  stories:
    padding: 4

#-----------------------------------------------------
connect = ReactRedux.connect mapStateToProps
module.exports = connect App
