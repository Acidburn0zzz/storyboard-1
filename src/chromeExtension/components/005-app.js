/* eslint-env browser */

/* eslint-disable react/no-string-refs */

import React from 'react';
import * as ReactRedux from 'react-redux';
import {
  Floats, Notifications,
  LargeMessage,
  Spinner,
  isDark,
} from 'giu';
import tinycolor from 'tinycolor2';
import { pick } from '../../vendor/lodash';
import Toolbar from './015-toolbar';
import Story from './020-story';
// let ReduxDevTools;
// if process.env.NODE_ENV isnt 'production' {
//   ReduxDevTools = require('../components/990-reduxDevTools')
// }

require('./app.sass');

const mapStateToProps = (state) => ({
  fRelativeTime: state.settings.timeType === 'RELATIVE',
  cxState: state.cx.cxState,
  mainStory: state.stories.mainStory,
  colors: pick(state.settings, [
    'colorClientBg', 'colorServerBg', 'colorUiBg',
    'colorClientFg', 'colorServerFg', 'colorUiFg',
    'colorClientBgIsDark', 'colorServerBgIsDark', 'colorUiBgIsDark',
  ]),
});

class App extends React.PureComponent {
  static propTypes = {
    // From Redux.connect
    fRelativeTime: React.PropTypes.bool.isRequired,
    cxState: React.PropTypes.string.isRequired,
    mainStory: React.PropTypes.object.isRequired,
    colors: React.PropTypes.object.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      seqFullRefresh: 0,
    };
  }

  componentDidMount() {
    this.timerFullRefresh = setInterval(this.fullRefresh, 30e3);
    window.addEventListener('scroll', this.onScroll);
  }

  componentWillUnmount() {
    clearInterval(this.timerFullRefresh);
    this.timerFullRefresh = null;
    window.removeEventListener('scroll', this.onScroll);
  }

  componentDidUpdate() {
    if (this.fAnchoredToBottom) {
      window.scrollTo(0, document.body.scrollHeight);
    }
  }

  // -----------------------------------------------------
  fullRefresh = () => {
    if (!this.props.fRelativeTime) return;
    this.setState({
      seqFullRefresh: this.state.seqFullRefresh + 1,
    });
  }

  onScroll = () => {
    const bcr = this.refs.outer.getBoundingClientRect();
    this.fAnchoredToBottom = bcr.bottom - window.innerHeight < 30;
  }

  // -----------------------------------------------------
  render() {
    let reduxDevTools;
    const { cxState, colors } = this.props;
    // if (process.env.NODE_ENV isnt 'production') {
    //   reduxDevTools = <ReduxDevTools />
    // }
    const fConnected = cxState === 'CONNECTED';
    return (
      <div ref="outer" id="appRoot" style={style.outer(colors)}>
        <Floats />
        <Notifications />
        {!fConnected && this.renderConnecting()}
        {fConnected && <Toolbar colors={colors} />}
        {fConnected && this.renderStories()}
        {reduxDevTools}
      </div>
    );
  }

  renderStories() {
    return (
      <div style={style.stories}>
        <Story
          story={this.props.mainStory}
          level={0}
          seqFullRefresh={this.state.seqFullRefresh}
          colors={this.props.colors}
        />
      </div>
    );
  }

  renderConnecting() {
    return (
      <LargeMessage style={style.largeMessage(this.props.colors)}>
        <div><Spinner /> Connecting to Storyboard...</div>{' '}
        <div>Navigate to your Storyboard-equipped app (and log in if required)</div>
      </LargeMessage>
    );
  }
}

// -----------------------------------------------------
const style = {
  outer: (colors) => ({
    height: '100%',
    backgroundColor: colors.colorUiBg,
    color: colors.colorUiFg,
  }),
  largeMessage: (colors) => {
    let color = tinycolor(colors.colorUiFg);
    color = isDark(colors.colorUiFg) ? color.lighten(20) : color.darken(20);
    return { color: color.toRgbString() };
  },
  stories: { padding: 4 },
};

// -----------------------------------------------------
const connect = ReactRedux.connect(mapStateToProps);
export default connect(App);
export { App as _App };
