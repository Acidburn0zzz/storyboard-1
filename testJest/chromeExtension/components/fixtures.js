import { addDefaults } from 'timm';

export const BASE_COLORS = {
  colorClientBg: 'aliceblue',
  colorClientBgIsDark: false,
  colorClientFg: 'black',
  colorServerBg: 'rgb(214, 236, 255)',
  colorServerBgIsDark: false,
  colorServerFg: 'black',
  colorUiBg: 'white',
  colorUiBgIsDark: false,
  colorUiFg: 'rgb(64, 64, 64)',
};

export const DARK_COLORS = {
  colorClientBg: 'rgb(17, 22, 54)',
  colorClientBgIsDark: true,
  colorClientFg: 'white',
  colorServerBg: 'rgb(14, 11, 33)',
  colorServerBgIsDark: true,
  colorServerFg: 'white',
  colorUiBg: 'black',
  colorUiBgIsDark: true,
  colorUiFg: 'white',
};

export const EMPTY_MAIN_STORY = {
  fExpanded: true,
  fHierarchical: true,
  fOpen: true,
  fWrapper: true,
  records: [
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: false,
      fStoryObject: true,
      id: 'main_0',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/0',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_0',
      t: 0,
      title: 'Client',
    },
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: true,
      fStoryObject: true,
      id: 'main_1',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/1',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_1',
      t: 0,
      title: 'Server',
    },
  ],
};

export const buildLogRecord = (options = {}) => addDefaults(options, {
  id: 'id1',
  hubId: 'hubId1',
  version: 3,
  fStory: false,
  fServer: false,
  storyId: 'story1',
  t: 0,
  src: 'main',
  level: 30,
  msg: 'Example message',
});

export const buildStory = (options = {}) => {
  const story = addDefaults(options, {
    id: 'id1',
    hubId: 'hubId1',
    version: 3,
    fStory: true,
    fServer: false,
    storyId: 'story1',
    t: 0,
    src: 'main',
    level: 30,
    fRoot: false,
    title: 'Story title',
    parents: [],
    // client-side
    fStoryObject: true,
    fOpen: false,
    fMain: false,
    status: undefined,
    fExpanded: true,
    fHierarchical: true,
    records: [],
  });
  story.records.unshift({
    id: `${story.id}.creation`,
    hubId: story.hubId,
    version: story.version,
    fStory: true,
    fServer: story.fServer,
    storyId: story.storyId,
    t: story.t,
    src: story.src,
    level: story.level,
    fRoot: story.fRoot,
    title: story.title,
    action: 'CREATED',
  });
  if (!story.fOpen) {
    story.records.push({
      id: `${story.id}.closure`,
      hubId: story.hubId,
      version: story.version,
      fStory: true,
      fServer: story.fServer,
      storyId: story.storyId,
      t: story.records[story.records.length - 1].t,
      src: story.src,
      level: story.level,
      fRoot: story.fRoot,
      title: story.title,
      action: 'CLOSED',
    });
  }
  story.numRecords = story.records.length;
  return story;
};
