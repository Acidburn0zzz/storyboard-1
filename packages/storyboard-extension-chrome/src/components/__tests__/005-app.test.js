/* eslint-env jest */
/* eslint-disable no-extend-native, global-require */
import React from 'react';
import renderer from 'react-test-renderer';
import { _App as App } from '../005-app';
import { BASE_COLORS } from './fixtures';

jest.mock('react-dom');
jest.mock('../015-toolbar', () => require('./mockComponent')('Toolbar'));
jest.mock('../020-story', () => require('./mockComponent')('Story'));

describe('App', () => {
  it('renders correctly when disconnected', () => {
    const tree = renderer.create(
      <App
        fRelativeTime={false}
        cxState="DISCONNECTED"
        mainStory={{ placeholderFor: 'mainStory' }}
        colors={BASE_COLORS}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when connected', () => {
    const tree = renderer.create(
      <App
        fRelativeTime={false}
        cxState="CONNECTED"
        mainStory={{ placeholderFor: 'mainStory' }}
        colors={BASE_COLORS}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
