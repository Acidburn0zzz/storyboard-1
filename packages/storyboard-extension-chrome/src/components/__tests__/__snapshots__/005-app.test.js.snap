exports[`App renders correctly when connected 1`] = `
<div
  id="appRoot"
  style={
    Object {
      "backgroundColor": "white",
      "color": "rgb(64, 64, 64)",
      "height": "100%",
    }
  }>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    } />
  <div
    className="giu-notifications"
    style={
      Object {
        "bottom": 20,
        "maxWidth": 350,
        "position": "fixed",
        "right": 20,
        "zIndex": 9000,
      }
    } />
  <div
    className="giu-hints">
    <div
      className="giu-hint-screen"
      onClick={[Function]}
      style={
        Object {
          "height": "100vh",
          "left": 0,
          "position": "fixed",
          "top": 0,
          "width": "100vw",
          "zIndex": 50,
        }
      }>
      <div
        className="giu-backdrop"
        onTouchMove={[Function]}
        onWheel={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "cursor": "pointer",
            "height": "100vh",
            "left": 0,
            "position": "fixed",
            "top": 0,
            "width": "100vw",
          }
        } />
      <div
        style={
          Object {
            "color": "white",
            "fontFamily": "\"Gloria Hallelujah\", sans-serif",
            "fontSize": 20,
          }
        }>
        <svg
          style={
            Object {
              "fill": "transparent",
              "height": "100vh",
              "pointerEvents": "none",
              "position": "fixed",
              "stroke": "white",
              "strokeWidth": 2,
              "width": "100vw",
            }
          }>
          <path
            className="giu-hint-arrow"
            d="M45,230 C22.25,235.25,14.75,247.25,20,270 M20,270 l2.9246155276374792,-9.562772820448144 M20,270 l-6.819296429308718,-7.3141777534605605" />
        </svg>
        <div
          className="giu-hint-label"
          style={
            Object {
              "left": 50,
              "pointerEvents": "none",
              "position": "fixed",
              "top": 230,
              "width": 0,
            }
          }>
          <div
            style={
              Object {
                "fontSize": 20,
                "left": 0,
                "lineHeight": "25px",
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": -14,
                "position": "absolute",
                "textAlign": "left",
                "width": 2000,
              }
            }>
            <span>
              <span
                style={
                  Object {
                    "color": "#4df950",
                  }
                }>
                Click
              </span>
               on timestamps to toggle: local, UTC, relative to now
              <br />
              <span
                style={
                  Object {
                    "color": "#4df950",
                  }
                }>
                Right-click
              </span>
               to set a reference timestamp
            </span>
          </div>
        </div>
        <div
          style={
            Object {
              "border": "1px solid white",
              "borderRadius": 8,
              "bottom": 20,
              "cursor": "pointer",
              "padding": "5px 25px",
              "position": "fixed",
              "right": 20,
            }
          }>
          Enjoy!
        </div>
      </div>
    </div>
  </div>
  <div
    colors={
      Object {
        "colorClientBg": "aliceblue",
        "colorClientBgIsDark": false,
        "colorClientFg": "black",
        "colorServerBg": "rgb(214, 236, 255)",
        "colorServerBgIsDark": false,
        "colorServerFg": "black",
        "colorUiBg": "white",
        "colorUiBgIsDark": false,
        "colorUiFg": "rgb(64, 64, 64)",
      }
    }
    dataMockType="Toolbar">
    <div
      style={
        Object {
          "color": "blue",
          "fontWeight": "bold",
        }
      }>
      Toolbar
    </div>
  </div>
  <div
    style={
      Object {
        "padding": 4,
      }
    }>
    <div
      colors={
        Object {
          "colorClientBg": "aliceblue",
          "colorClientBgIsDark": false,
          "colorClientFg": "black",
          "colorServerBg": "rgb(214, 236, 255)",
          "colorServerBgIsDark": false,
          "colorServerFg": "black",
          "colorUiBg": "white",
          "colorUiBgIsDark": false,
          "colorUiFg": "rgb(64, 64, 64)",
        }
      }
      dataMockType="Story"
      level={0}
      seqFullRefresh={0}
      setTimeRef={[Function]}
      story={
        Object {
          "placeholderFor": "mainStory",
        }
      }
      timeRef={null}>
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }>
        Story
      </div>
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div
  id="appRoot"
  style="background-color:white;color:rgb(64, 64, 64);height:100%;"
>
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div
    class="giu-notifications"
    style="bottom:20px;max-width:350px;position:fixed;right:20px;z-index:9000;"
  >
  </div>
  <div class="giu-hints">
    <div
      class="giu-hint-screen"
      style="height:100vh;left:0;position:fixed;top:0;width:100vw;z-index:50;"
    >
      <div
        class="giu-backdrop"
        style="background-color:black;cursor:pointer;height:100vh;left:0;position:fixed;top:0;width:100vw;"
      >
      </div>
      <div style="color:white;font-family:\"Gloria Hallelujah\", sans-serif;font-size:20px;">
        <svg style="fill:transparent;height:100vh;pointer-events:none;position:fixed;stroke:white;stroke-width:2;width:100vw;">
          <path
            class="giu-hint-arrow"
            d="M45,230 C22.25,235.25,14.75,247.25,20,270 M20,270 l2.9246155276374792,-9.562772820448144 M20,270 l-6.819296429308718,-7.3141777534605605"
          >
          </path>
        </svg>
        <div
          class="giu-hint-label"
          style="left:50px;pointer-events:none;position:fixed;top:230px;width:0;"
        >
          <div style="font-size:20px;left:0;line-height:25px;margin-left:8px;margin-right:8px;margin-top:-14px;position:absolute;text-align:left;width:2000px;">
            <span>
              <span style="color:#4df950;">
                Click
              </span>
               on timestamps to toggle: local, UTC, relative to now
              <br>
              <span style="color:#4df950;">
                Right-click
              </span>
               to set a reference timestamp
            </span>
          </div>
        </div>
        <div style="border:1px solid white;border-radius:8px;bottom:20px;cursor:pointer;padding:5px 25px;position:fixed;right:20px;">
          Enjoy!
        </div>
      </div>
    </div>
  </div>
  <div dataMockType="Toolbar">
    <div style="color:blue;font-weight:bold;">
      Toolbar
    </div>
  </div>
  <div style="padding:4px;">
    <div dataMockType="Story">
      <div style="color:blue;font-weight:bold;">
        Story
      </div>
    </div>
  </div>
</div>
`;

exports[`App renders correctly when disconnected 1`] = `
<div
  id="appRoot"
  style={
    Object {
      "backgroundColor": "white",
      "color": "rgb(64, 64, 64)",
      "height": "100%",
    }
  }>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    } />
  <div
    className="giu-notifications"
    style={
      Object {
        "bottom": 20,
        "maxWidth": 350,
        "position": "fixed",
        "right": 20,
        "zIndex": 9000,
      }
    } />
  <div
    className="giu-hints" />
  <div
    className="giu-large-message"
    style={
      Object {
        "color": "rgb(115, 115, 115)",
        "fontSize": "1.4em",
        "fontWeight": 700,
        "padding": "0.8em",
        "textAlign": "center",
      }
    }>
    <div>
      <i
        className="giu-icon fa fa-circle-o-notch fa-spin"
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "letterSpacing": "normal",
          }
        } />
       Connecting to Storyboard...
    </div>
     
    <div>
      Navigate to your Storyboard-equipped app (and log in if required)
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div
  id="appRoot"
  style="background-color:white;color:rgb(64, 64, 64);height:100%;"
>
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div
    class="giu-notifications"
    style="bottom:20px;max-width:350px;position:fixed;right:20px;z-index:9000;"
  >
  </div>
  <div class="giu-hints"></div>
  <div
    class="giu-large-message"
    style="color:rgb(115, 115, 115);font-size:1.4em;font-weight:700;padding:0.8em;text-align:center;"
  >
    <div>
      <i
        class="giu-icon fa fa-circle-o-notch fa-spin"
        style="cursor:pointer;letter-spacing:normal;"
      >
      </i>
       Connecting to Storyboard...
    </div>
     
    <div>
      Navigate to your Storyboard-equipped app (and log in if required)
    </div>
  </div>
</div>
`;
